### ISSUE TITLE
Activity deletion not supported

### ISSUE THREAD

**Issue by tmuras** [pmaneggia/moodle-block_availdep#5]
After deleting activity from course, it stays in the diagram.

**Comment by pmaneggia** [comment]
Many thanks for the feedback, I will implement this!

**Comment by pmaneggia** [comment]
I had underestimated the problem. It is not the case that the graph needs to be computed anew when deleting an activity... it is much more that the condition completion itself does not really support deletion of an activity? (only the case "previous activity" completion seems to be able to handle that). In this case to support deleting an activity I need to do the job myself:

1. check if a course module has the flag `deletioninprogress` set and in this case remove the affected nodes and edges from the graph (or display them differently);
2. once the course module has been removed by the ad hoc task... handle somehow dependencies on course modules that are not to be found anymore (the json string in the field `availability` is apparently not going to be changed) in a similar way to the string that states the availability conditions.
